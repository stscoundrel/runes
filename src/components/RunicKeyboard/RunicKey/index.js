import styles from './RunicKey.module.scss'

export default function RunicKey({ rune, letter, callback }) {
  return (
    <button
      className={styles.key}
      aria-label={`Runic key ${rune}, meaning letter ${letter}`}
      onClick={() => callback(letter)}
    >
      {rune}
    </button>
  )
}
