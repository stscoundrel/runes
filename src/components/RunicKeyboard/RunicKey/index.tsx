import styles from './RunicKey.module.scss'

interface RunicKeyCallback {
  (param: unknown): void
}

interface RunicKeyProps {
  rune: string
  letter: string
  callback: RunicKeyCallback
}

export default function RunicKey({ rune, letter, callback }: RunicKeyProps) {
  return (
    <button
      title={`Runic key ${rune}, meaning letter ${letter}`}
      className={styles.key}
      aria-label={`Runic key ${rune}, meaning letter ${letter}`}
      onClick={() => callback(letter)}
    >
      {rune}
    </button>
  )
}
