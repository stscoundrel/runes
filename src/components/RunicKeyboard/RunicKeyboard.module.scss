.keyboard {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}

.result {
  display: block;
  min-height: rem(40px);
  font-style: italic;
}