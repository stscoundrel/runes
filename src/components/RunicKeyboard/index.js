import { useState } from 'react'
import RunicKey from 'components/RunicKeyboard/RunicKey'

export default function RunicKeyboard({ dialect, runeSet }) {
  const [text, setText] = useState('')

  const appendLetter = (letter) => {
    setText(`${text}${letter}`)
  }

  return (
    <section>
      <header>
        <h1>{dialect}</h1>
        <p>Translate runes to text</p>
      </header>

      <p>{text}</p>

      {Array.from(runeSet).map(([rune, letter]) => (
        <RunicKey key={rune} rune={rune} letter={letter} callback={appendLetter} />
      ))}

    </section>
  )
}
