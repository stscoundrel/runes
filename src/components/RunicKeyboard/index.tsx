import { useState } from 'react'
import RunicKey from 'components/RunicKeyboard/RunicKey'
import styles from './RunicKeyboard.module.scss'

interface RunicKeyboardProps {
  dialect: string
  runeSet: Map<string, string>
}

export default function RunicKeyboard({ dialect, runeSet }: RunicKeyboardProps) {
  const [text, setText] = useState('')

  const appendLetter = (letter: string): void => {
    setText(`${text}${letter}`)
  }

  return (
    <section>
      <header>
        <h1>{dialect}</h1>
        <p>Translate runes to text</p>
      </header>

      <p className={styles.result}>{text}</p>

      <div className={styles.keyboard}>
        {Array.from(runeSet).map(([rune, letter]) => (
          <RunicKey key={rune} rune={rune} letter={letter} callback={appendLetter} />
        ))}
      </div>

    </section>
  )
}
