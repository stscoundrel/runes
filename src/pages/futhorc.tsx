// Libs.
import futhorc from 'futhorc'

// Components.
import Layout from 'components/Layout'
import RunicKeyboard from 'components/RunicKeyboard'

export default function Futhork() {
  return (
    <Layout>
      <RunicKeyboard dialect="Futhorc" runeSet={futhorc.getRuneMapping()} />
    </Layout>
  )
}
