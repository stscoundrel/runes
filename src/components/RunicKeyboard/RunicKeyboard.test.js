import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

// Rune adapters.
import youngerFuthark from 'younger-futhark'
import elderFuthark from 'elder-futhark'
import futhorc from 'futhorc'

// Components.
import RunicKeyboard from './index'

describe('RunicKey component', () => {
  const elderFutharkRuneSet = elderFuthark.getRuneMapping()
  const youngerFutharkRuneSet = youngerFuthark.getRuneMapping()
  const futhorcRuneSet = futhorc.getRuneMapping()

  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<RunicKeyboard dialect="Some runic dialect" runeSet={futhorcRuneSet}/>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot: Elder Futhark', () => {
    const tree = renderer.create(<RunicKeyboard dialect="Elder Futhark" runeSet={elderFutharkRuneSet}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Matches snapshot: Younger Futhark', () => {
    const tree = renderer.create(<RunicKeyboard dialect="Younger Futhark" runeSet={youngerFutharkRuneSet}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Matches snapshot: Futhorc', () => {
    const tree = renderer.create(<RunicKeyboard dialect="Futhorc" runeSet={futhorcRuneSet}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
