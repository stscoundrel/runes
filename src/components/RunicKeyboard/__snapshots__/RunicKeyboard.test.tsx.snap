// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunicKey component Matches snapshot: Elder Futhark 1`] = `
<section>
  <header>
    <h1>
      Elder Futhark
    </h1>
    <p>
      Translate runes to text
    </p>
  </header>
  <p
    className="result"
  />
  <div
    className="keyboard"
  >
    <button
      aria-label="Runic key ᚠ, meaning letter f"
      className="key"
      onClick={[Function]}
      title="Runic key ᚠ, meaning letter f"
    >
      ᚠ
    </button>
    <button
      aria-label="Runic key ᚢ, meaning letter u"
      className="key"
      onClick={[Function]}
      title="Runic key ᚢ, meaning letter u"
    >
      ᚢ
    </button>
    <button
      aria-label="Runic key ᚦ, meaning letter þ"
      className="key"
      onClick={[Function]}
      title="Runic key ᚦ, meaning letter þ"
    >
      ᚦ
    </button>
    <button
      aria-label="Runic key ᚨ, meaning letter a"
      className="key"
      onClick={[Function]}
      title="Runic key ᚨ, meaning letter a"
    >
      ᚨ
    </button>
    <button
      aria-label="Runic key ᚱ, meaning letter r"
      className="key"
      onClick={[Function]}
      title="Runic key ᚱ, meaning letter r"
    >
      ᚱ
    </button>
    <button
      aria-label="Runic key ᚲ, meaning letter k"
      className="key"
      onClick={[Function]}
      title="Runic key ᚲ, meaning letter k"
    >
      ᚲ
    </button>
    <button
      aria-label="Runic key ᚷ, meaning letter g"
      className="key"
      onClick={[Function]}
      title="Runic key ᚷ, meaning letter g"
    >
      ᚷ
    </button>
    <button
      aria-label="Runic key ᚹ, meaning letter w"
      className="key"
      onClick={[Function]}
      title="Runic key ᚹ, meaning letter w"
    >
      ᚹ
    </button>
    <button
      aria-label="Runic key ᚺ, meaning letter h"
      className="key"
      onClick={[Function]}
      title="Runic key ᚺ, meaning letter h"
    >
      ᚺ
    </button>
    <button
      aria-label="Runic key ᚻ, meaning letter h"
      className="key"
      onClick={[Function]}
      title="Runic key ᚻ, meaning letter h"
    >
      ᚻ
    </button>
    <button
      aria-label="Runic key ᚾ, meaning letter n"
      className="key"
      onClick={[Function]}
      title="Runic key ᚾ, meaning letter n"
    >
      ᚾ
    </button>
    <button
      aria-label="Runic key ᛁ, meaning letter i"
      className="key"
      onClick={[Function]}
      title="Runic key ᛁ, meaning letter i"
    >
      ᛁ
    </button>
    <button
      aria-label="Runic key ᛃ, meaning letter j"
      className="key"
      onClick={[Function]}
      title="Runic key ᛃ, meaning letter j"
    >
      ᛃ
    </button>
    <button
      aria-label="Runic key ᛇ, meaning letter ï"
      className="key"
      onClick={[Function]}
      title="Runic key ᛇ, meaning letter ï"
    >
      ᛇ
    </button>
    <button
      aria-label="Runic key ᛈ, meaning letter p"
      className="key"
      onClick={[Function]}
      title="Runic key ᛈ, meaning letter p"
    >
      ᛈ
    </button>
    <button
      aria-label="Runic key ᛉ, meaning letter z"
      className="key"
      onClick={[Function]}
      title="Runic key ᛉ, meaning letter z"
    >
      ᛉ
    </button>
    <button
      aria-label="Runic key ᛊ, meaning letter s"
      className="key"
      onClick={[Function]}
      title="Runic key ᛊ, meaning letter s"
    >
      ᛊ
    </button>
    <button
      aria-label="Runic key ᛋ, meaning letter s"
      className="key"
      onClick={[Function]}
      title="Runic key ᛋ, meaning letter s"
    >
      ᛋ
    </button>
    <button
      aria-label="Runic key ᛏ, meaning letter t"
      className="key"
      onClick={[Function]}
      title="Runic key ᛏ, meaning letter t"
    >
      ᛏ
    </button>
    <button
      aria-label="Runic key ᛒ, meaning letter b"
      className="key"
      onClick={[Function]}
      title="Runic key ᛒ, meaning letter b"
    >
      ᛒ
    </button>
    <button
      aria-label="Runic key ᛖ, meaning letter e"
      className="key"
      onClick={[Function]}
      title="Runic key ᛖ, meaning letter e"
    >
      ᛖ
    </button>
    <button
      aria-label="Runic key ᛗ, meaning letter m"
      className="key"
      onClick={[Function]}
      title="Runic key ᛗ, meaning letter m"
    >
      ᛗ
    </button>
    <button
      aria-label="Runic key ᛚ, meaning letter l"
      className="key"
      onClick={[Function]}
      title="Runic key ᛚ, meaning letter l"
    >
      ᛚ
    </button>
    <button
      aria-label="Runic key ᛜ, meaning letter ŋ"
      className="key"
      onClick={[Function]}
      title="Runic key ᛜ, meaning letter ŋ"
    >
      ᛜ
    </button>
    <button
      aria-label="Runic key ᛝ, meaning letter ŋ"
      className="key"
      onClick={[Function]}
      title="Runic key ᛝ, meaning letter ŋ"
    >
      ᛝ
    </button>
    <button
      aria-label="Runic key ᛟ, meaning letter o"
      className="key"
      onClick={[Function]}
      title="Runic key ᛟ, meaning letter o"
    >
      ᛟ
    </button>
    <button
      aria-label="Runic key ᛞ, meaning letter d"
      className="key"
      onClick={[Function]}
      title="Runic key ᛞ, meaning letter d"
    >
      ᛞ
    </button>
    <button
      aria-label="Runic key :, meaning letter  "
      className="key"
      onClick={[Function]}
      title="Runic key :, meaning letter  "
    >
      :
    </button>
  </div>
</section>
`;

exports[`RunicKey component Matches snapshot: Futhorc 1`] = `
<section>
  <header>
    <h1>
      Futhorc
    </h1>
    <p>
      Translate runes to text
    </p>
  </header>
  <p
    className="result"
  />
  <div
    className="keyboard"
  >
    <button
      aria-label="Runic key ᚠ, meaning letter f"
      className="key"
      onClick={[Function]}
      title="Runic key ᚠ, meaning letter f"
    >
      ᚠ
    </button>
    <button
      aria-label="Runic key ᚢ, meaning letter u"
      className="key"
      onClick={[Function]}
      title="Runic key ᚢ, meaning letter u"
    >
      ᚢ
    </button>
    <button
      aria-label="Runic key ᚦ, meaning letter þ"
      className="key"
      onClick={[Function]}
      title="Runic key ᚦ, meaning letter þ"
    >
      ᚦ
    </button>
    <button
      aria-label="Runic key ᚩ, meaning letter o"
      className="key"
      onClick={[Function]}
      title="Runic key ᚩ, meaning letter o"
    >
      ᚩ
    </button>
    <button
      aria-label="Runic key ᚱ, meaning letter r"
      className="key"
      onClick={[Function]}
      title="Runic key ᚱ, meaning letter r"
    >
      ᚱ
    </button>
    <button
      aria-label="Runic key ᚳ, meaning letter c"
      className="key"
      onClick={[Function]}
      title="Runic key ᚳ, meaning letter c"
    >
      ᚳ
    </button>
    <button
      aria-label="Runic key ᚷ, meaning letter g"
      className="key"
      onClick={[Function]}
      title="Runic key ᚷ, meaning letter g"
    >
      ᚷ
    </button>
    <button
      aria-label="Runic key ᚹ, meaning letter w"
      className="key"
      onClick={[Function]}
      title="Runic key ᚹ, meaning letter w"
    >
      ᚹ
    </button>
    <button
      aria-label="Runic key ᚻ, meaning letter h"
      className="key"
      onClick={[Function]}
      title="Runic key ᚻ, meaning letter h"
    >
      ᚻ
    </button>
    <button
      aria-label="Runic key ᚾ, meaning letter n"
      className="key"
      onClick={[Function]}
      title="Runic key ᚾ, meaning letter n"
    >
      ᚾ
    </button>
    <button
      aria-label="Runic key ᛁ, meaning letter i"
      className="key"
      onClick={[Function]}
      title="Runic key ᛁ, meaning letter i"
    >
      ᛁ
    </button>
    <button
      aria-label="Runic key ᛡ, meaning letter j"
      className="key"
      onClick={[Function]}
      title="Runic key ᛡ, meaning letter j"
    >
      ᛡ
    </button>
    <button
      aria-label="Runic key ᛄ, meaning letter j"
      className="key"
      onClick={[Function]}
      title="Runic key ᛄ, meaning letter j"
    >
      ᛄ
    </button>
    <button
      aria-label="Runic key ᛇ, meaning letter ï"
      className="key"
      onClick={[Function]}
      title="Runic key ᛇ, meaning letter ï"
    >
      ᛇ
    </button>
    <button
      aria-label="Runic key ᛈ, meaning letter p"
      className="key"
      onClick={[Function]}
      title="Runic key ᛈ, meaning letter p"
    >
      ᛈ
    </button>
    <button
      aria-label="Runic key ᛉ, meaning letter x"
      className="key"
      onClick={[Function]}
      title="Runic key ᛉ, meaning letter x"
    >
      ᛉ
    </button>
    <button
      aria-label="Runic key ᛋ, meaning letter s"
      className="key"
      onClick={[Function]}
      title="Runic key ᛋ, meaning letter s"
    >
      ᛋ
    </button>
    <button
      aria-label="Runic key ᚴ, meaning letter s"
      className="key"
      onClick={[Function]}
      title="Runic key ᚴ, meaning letter s"
    >
      ᚴ
    </button>
    <button
      aria-label="Runic key ᛏ, meaning letter t"
      className="key"
      onClick={[Function]}
      title="Runic key ᛏ, meaning letter t"
    >
      ᛏ
    </button>
    <button
      aria-label="Runic key ᛒ, meaning letter b"
      className="key"
      onClick={[Function]}
      title="Runic key ᛒ, meaning letter b"
    >
      ᛒ
    </button>
    <button
      aria-label="Runic key ᛖ, meaning letter e"
      className="key"
      onClick={[Function]}
      title="Runic key ᛖ, meaning letter e"
    >
      ᛖ
    </button>
    <button
      aria-label="Runic key ᛗ, meaning letter m"
      className="key"
      onClick={[Function]}
      title="Runic key ᛗ, meaning letter m"
    >
      ᛗ
    </button>
    <button
      aria-label="Runic key ᛚ, meaning letter l"
      className="key"
      onClick={[Function]}
      title="Runic key ᛚ, meaning letter l"
    >
      ᛚ
    </button>
    <button
      aria-label="Runic key ᛝ, meaning letter ŋ"
      className="key"
      onClick={[Function]}
      title="Runic key ᛝ, meaning letter ŋ"
    >
      ᛝ
    </button>
    <button
      aria-label="Runic key ᛟ, meaning letter œ"
      className="key"
      onClick={[Function]}
      title="Runic key ᛟ, meaning letter œ"
    >
      ᛟ
    </button>
    <button
      aria-label="Runic key ᛞ, meaning letter d"
      className="key"
      onClick={[Function]}
      title="Runic key ᛞ, meaning letter d"
    >
      ᛞ
    </button>
    <button
      aria-label="Runic key ᚪ, meaning letter a"
      className="key"
      onClick={[Function]}
      title="Runic key ᚪ, meaning letter a"
    >
      ᚪ
    </button>
    <button
      aria-label="Runic key ᚫ, meaning letter æ"
      className="key"
      onClick={[Function]}
      title="Runic key ᚫ, meaning letter æ"
    >
      ᚫ
    </button>
    <button
      aria-label="Runic key ᚣ, meaning letter y"
      className="key"
      onClick={[Function]}
      title="Runic key ᚣ, meaning letter y"
    >
      ᚣ
    </button>
    <button
      aria-label="Runic key ᛠ, meaning letter ea"
      className="key"
      onClick={[Function]}
      title="Runic key ᛠ, meaning letter ea"
    >
      ᛠ
    </button>
    <button
      aria-label="Runic key :, meaning letter  "
      className="key"
      onClick={[Function]}
      title="Runic key :, meaning letter  "
    >
      :
    </button>
  </div>
</section>
`;

exports[`RunicKey component Matches snapshot: Futhork 1`] = `
<section>
  <header>
    <h1>
      Futhork
    </h1>
    <p>
      Translate runes to text
    </p>
  </header>
  <p
    className="result"
  />
  <div
    className="keyboard"
  >
    <button
      aria-label="Runic key ᚠ, meaning letter f"
      className="key"
      onClick={[Function]}
      title="Runic key ᚠ, meaning letter f"
    >
      ᚠ
    </button>
    <button
      aria-label="Runic key ᚢ, meaning letter u"
      className="key"
      onClick={[Function]}
      title="Runic key ᚢ, meaning letter u"
    >
      ᚢ
    </button>
    <button
      aria-label="Runic key ᚦ, meaning letter þ"
      className="key"
      onClick={[Function]}
      title="Runic key ᚦ, meaning letter þ"
    >
      ᚦ
    </button>
    <button
      aria-label="Runic key ᚮ, meaning letter o"
      className="key"
      onClick={[Function]}
      title="Runic key ᚮ, meaning letter o"
    >
      ᚮ
    </button>
    <button
      aria-label="Runic key ᚱ, meaning letter r"
      className="key"
      onClick={[Function]}
      title="Runic key ᚱ, meaning letter r"
    >
      ᚱ
    </button>
    <button
      aria-label="Runic key ᚴ, meaning letter k"
      className="key"
      onClick={[Function]}
      title="Runic key ᚴ, meaning letter k"
    >
      ᚴ
    </button>
    <button
      aria-label="Runic key ᚼ, meaning letter h"
      className="key"
      onClick={[Function]}
      title="Runic key ᚼ, meaning letter h"
    >
      ᚼ
    </button>
    <button
      aria-label="Runic key ᚿ, meaning letter n"
      className="key"
      onClick={[Function]}
      title="Runic key ᚿ, meaning letter n"
    >
      ᚿ
    </button>
    <button
      aria-label="Runic key ᛁ, meaning letter i"
      className="key"
      onClick={[Function]}
      title="Runic key ᛁ, meaning letter i"
    >
      ᛁ
    </button>
    <button
      aria-label="Runic key ᛆ, meaning letter a"
      className="key"
      onClick={[Function]}
      title="Runic key ᛆ, meaning letter a"
    >
      ᛆ
    </button>
    <button
      aria-label="Runic key ᛌ, meaning letter s"
      className="key"
      onClick={[Function]}
      title="Runic key ᛌ, meaning letter s"
    >
      ᛌ
    </button>
    <button
      aria-label="Runic key ᛋ, meaning letter s"
      className="key"
      onClick={[Function]}
      title="Runic key ᛋ, meaning letter s"
    >
      ᛋ
    </button>
    <button
      aria-label="Runic key ᛐ, meaning letter t"
      className="key"
      onClick={[Function]}
      title="Runic key ᛐ, meaning letter t"
    >
      ᛐ
    </button>
    <button
      aria-label="Runic key ᛏ, meaning letter t"
      className="key"
      onClick={[Function]}
      title="Runic key ᛏ, meaning letter t"
    >
      ᛏ
    </button>
    <button
      aria-label="Runic key ᛒ, meaning letter b"
      className="key"
      onClick={[Function]}
      title="Runic key ᛒ, meaning letter b"
    >
      ᛒ
    </button>
    <button
      aria-label="Runic key ᛘ, meaning letter m"
      className="key"
      onClick={[Function]}
      title="Runic key ᛘ, meaning letter m"
    >
      ᛘ
    </button>
    <button
      aria-label="Runic key ᛚ, meaning letter l"
      className="key"
      onClick={[Function]}
      title="Runic key ᛚ, meaning letter l"
    >
      ᛚ
    </button>
    <button
      aria-label="Runic key ᛦ, meaning letter y"
      className="key"
      onClick={[Function]}
      title="Runic key ᛦ, meaning letter y"
    >
      ᛦ
    </button>
    <button
      aria-label="Runic key :, meaning letter  "
      className="key"
      onClick={[Function]}
      title="Runic key :, meaning letter  "
    >
      :
    </button>
    <button
      aria-label="Runic key ᚵ, meaning letter g"
      className="key"
      onClick={[Function]}
      title="Runic key ᚵ, meaning letter g"
    >
      ᚵ
    </button>
    <button
      aria-label="Runic key ᚽ, meaning letter e"
      className="key"
      onClick={[Function]}
      title="Runic key ᚽ, meaning letter e"
    >
      ᚽ
    </button>
    <button
      aria-label="Runic key ᚯ, meaning letter ø"
      className="key"
      onClick={[Function]}
      title="Runic key ᚯ, meaning letter ø"
    >
      ᚯ
    </button>
    <button
      aria-label="Runic key ᛅ, meaning letter æ"
      className="key"
      onClick={[Function]}
      title="Runic key ᛅ, meaning letter æ"
    >
      ᛅ
    </button>
    <button
      aria-label="Runic key ᚰ, meaning letter ǫ"
      className="key"
      onClick={[Function]}
      title="Runic key ᚰ, meaning letter ǫ"
    >
      ᚰ
    </button>
    <button
      aria-label="Runic key ᛕ, meaning letter ᴘ"
      className="key"
      onClick={[Function]}
      title="Runic key ᛕ, meaning letter ᴘ"
    >
      ᛕ
    </button>
  </div>
</section>
`;

exports[`RunicKey component Matches snapshot: Younger Futhark 1`] = `
<section>
  <header>
    <h1>
      Younger Futhark
    </h1>
    <p>
      Translate runes to text
    </p>
  </header>
  <p
    className="result"
  />
  <div
    className="keyboard"
  >
    <button
      aria-label="Runic key ᚠ, meaning letter f"
      className="key"
      onClick={[Function]}
      title="Runic key ᚠ, meaning letter f"
    >
      ᚠ
    </button>
    <button
      aria-label="Runic key ᚢ, meaning letter u"
      className="key"
      onClick={[Function]}
      title="Runic key ᚢ, meaning letter u"
    >
      ᚢ
    </button>
    <button
      aria-label="Runic key ᚦ, meaning letter þ"
      className="key"
      onClick={[Function]}
      title="Runic key ᚦ, meaning letter þ"
    >
      ᚦ
    </button>
    <button
      aria-label="Runic key ᚬ, meaning letter o"
      className="key"
      onClick={[Function]}
      title="Runic key ᚬ, meaning letter o"
    >
      ᚬ
    </button>
    <button
      aria-label="Runic key ᚱ, meaning letter r"
      className="key"
      onClick={[Function]}
      title="Runic key ᚱ, meaning letter r"
    >
      ᚱ
    </button>
    <button
      aria-label="Runic key ᚴ, meaning letter k"
      className="key"
      onClick={[Function]}
      title="Runic key ᚴ, meaning letter k"
    >
      ᚴ
    </button>
    <button
      aria-label="Runic key ᚼ, meaning letter h"
      className="key"
      onClick={[Function]}
      title="Runic key ᚼ, meaning letter h"
    >
      ᚼ
    </button>
    <button
      aria-label="Runic key ᚽ, meaning letter h"
      className="key"
      onClick={[Function]}
      title="Runic key ᚽ, meaning letter h"
    >
      ᚽ
    </button>
    <button
      aria-label="Runic key ᚾ, meaning letter n"
      className="key"
      onClick={[Function]}
      title="Runic key ᚾ, meaning letter n"
    >
      ᚾ
    </button>
    <button
      aria-label="Runic key ᚿ, meaning letter n"
      className="key"
      onClick={[Function]}
      title="Runic key ᚿ, meaning letter n"
    >
      ᚿ
    </button>
    <button
      aria-label="Runic key ᛁ, meaning letter i"
      className="key"
      onClick={[Function]}
      title="Runic key ᛁ, meaning letter i"
    >
      ᛁ
    </button>
    <button
      aria-label="Runic key ᛅ, meaning letter a"
      className="key"
      onClick={[Function]}
      title="Runic key ᛅ, meaning letter a"
    >
      ᛅ
    </button>
    <button
      aria-label="Runic key ᛆ, meaning letter a"
      className="key"
      onClick={[Function]}
      title="Runic key ᛆ, meaning letter a"
    >
      ᛆ
    </button>
    <button
      aria-label="Runic key ᛋ, meaning letter s"
      className="key"
      onClick={[Function]}
      title="Runic key ᛋ, meaning letter s"
    >
      ᛋ
    </button>
    <button
      aria-label="Runic key ᛌ, meaning letter s"
      className="key"
      onClick={[Function]}
      title="Runic key ᛌ, meaning letter s"
    >
      ᛌ
    </button>
    <button
      aria-label="Runic key ᛏ, meaning letter t"
      className="key"
      onClick={[Function]}
      title="Runic key ᛏ, meaning letter t"
    >
      ᛏ
    </button>
    <button
      aria-label="Runic key ᛐ, meaning letter t"
      className="key"
      onClick={[Function]}
      title="Runic key ᛐ, meaning letter t"
    >
      ᛐ
    </button>
    <button
      aria-label="Runic key ᛒ, meaning letter b"
      className="key"
      onClick={[Function]}
      title="Runic key ᛒ, meaning letter b"
    >
      ᛒ
    </button>
    <button
      aria-label="Runic key ᛘ, meaning letter m"
      className="key"
      onClick={[Function]}
      title="Runic key ᛘ, meaning letter m"
    >
      ᛘ
    </button>
    <button
      aria-label="Runic key ᛚ, meaning letter l"
      className="key"
      onClick={[Function]}
      title="Runic key ᛚ, meaning letter l"
    >
      ᛚ
    </button>
    <button
      aria-label="Runic key ᛦ, meaning letter R"
      className="key"
      onClick={[Function]}
      title="Runic key ᛦ, meaning letter R"
    >
      ᛦ
    </button>
    <button
      aria-label="Runic key :, meaning letter  "
      className="key"
      onClick={[Function]}
      title="Runic key :, meaning letter  "
    >
      :
    </button>
  </div>
</section>
`;
